h/static/scripts/directive/group-list.js

```diff
@@ -1,20 +1,75 @@
 'use strict';
 
+// @ngInject
+function GroupsListController($scope) {
+  $scope.expandedGroupId = undefined;
+
+  // used to keep the dropdown from closing when the user
+  // interacts with the inline link share pane within the groups list
+  $scope.stopClickPropagation = function($event) {
+    $event.stopPropagation();
+  }
+
+  // show the share link for the specified group or clear it if
+  // null
+  $scope.toggleShareLink = function (groupId) {
+    if (!groupId || $scope.expandedGroupId === groupId) {
+      $scope.expandedGroupId = undefined;
+    } else {
+      $scope.expandedGroupId = groupId;
+    }
+  };
+
+  $scope.shouldShowShareLink = function (groupId) {
+    return $scope.expandedGroupId === groupId;
+  }
+
+  $scope.sortedGroups = function () {
+    return $scope.groups.all().concat().sort(function (a, b) {
+      if (a.public !== b.public) {
+        return a.public ? -1 : 1;
+      }
+      return a.name.localeCompare(b.name);
+    });
+  }
+}
+
 /**
  * @ngdoc directive
  * @name groupList
  * @restrict AE
  * @description Displays a list of groups of which the user is a member.
  */
-
 // @ngInject
-module.exports = function (groups) {
+function groupsList(groups, $window) {
```

Comment:
Everywhere else this directive is referred to as `groupList`, so both this and the controller should probably be `group*` (singular) rather than `groups*` (plural).